File: flex-chat/frontend/src/types/theme.ts
export interface ThemePropertyType {
    id: number;
    key: string;
    displayName: string;
    description: string;
    defaultValue: string;
    propertyType: string;
    isRequired: boolean;
    createdAt: string;
}

export interface ThemeProperty {
    id: number;
    themeId: number;
    propertyTypeId: number;
    propertyValue: string;
    updatedAt: string;
    themePropertyType: ThemePropertyType;
}

export interface ThemePropertyMinimal {
    [key:string]:string;
}

export interface Theme {
    id: number;
    userId: number;
    name: string;
    isDefault: boolean;
    properties: ThemePropertyMinimal;
    createdAt: string;
    updatedAt: string;
}

export interface CreateOrUpdateTheme {
    name: string;
    properties: Array<{
        propertyTypeId: number;
        propertyValue: string;
    }>;
}

export interface ApiResponse<T> {
    data: T;
    message: string;
}

export type GetThemeResponse = ApiResponse<Theme>;
export type GetThemesResponse = ApiResponse<Theme[]>;
export type CreateThemeResponse = ApiResponse<Theme>;
export type UpdateThemeResponse = ApiResponse<Theme>;

File: flex-chat/backend/FlexChat.Data/Models/Theme.cs
using FlexChat.Data.Models.Base;
using System;
using System.Collections.Generic;
using System.Text;

namespace FlexChat.Data.Models
{
    public class Theme : AuditableEntity<int>
    {
        public int UserId { get; set; }
        public string Name { get; set; }
        public bool IsDefault { get; set; }

        public User User { get; set; }
        public ICollection<ThemeProperty> Properties { get; set; } = new List<ThemeProperty>();
    }
}
